src/main/java/gui/viewmodel/TodoListPageViewModel.java

```diff
@@ -12,87 +13,78 @@
-        repository.addCreationObserver(this::handleCreation);
-        repository.addUpdateObserver(this::handleUpdate);
+    public TodoListPageViewModel(TodoListRequestBoundary taskGetter) {
 
-        // initialize sorted taskReaderList
-        List<TaskReader> taskList = repository.getAllTasks().get(0L);
-        this.taskReaderList = FXCollections.observableArrayList();
-        for (TaskReader taskReader : taskList) {
-            this.insertTaskReader(taskReader);
-        }
-        // initialize taskInfoList accordingly
+        // initialize sorted taskInfoList
+        List<TaskInfo> taskList = taskGetter.getTasks();
         this.taskInfoList = FXCollections.observableArrayList();
-        this.updateTaskInfoList();
+        for (TaskInfo taskInfo : taskList) {
+            this.insertTaskInfo(taskInfo);
+        }
+        // initialize viewInfoList accordingly
+        this.viewInfoList = FXCollections.observableArrayList();
+        this.updateViewInfoList();
     }
 
     /***
-     * helper function to sort the taskReaderList based on the deadline
-     * @param taskReader to inserted to taskReadList
+     * helper function to sort the taskInfoList based on the deadline
+     * @param taskInfo to inserted to taskInfoList
      */
-    private void insertTaskReader(TaskReader taskReader) {
-        // put the non-deadline task to the end of the list
-        if (taskReader.getDeadline() != null) {
-            LocalDateTime deadline = taskReader.getDeadline();
+    private void insertTaskInfo(TaskInfo taskInfo) {
+        // put tasks with no deadlines to the end of the list
+        if (taskInfo.getDeadline() != null) {
+            LocalDateTime deadline = taskInfo.getDeadline();
             int i = 0;
-            while (i < this.taskReaderList.size()) {
-                if (taskReaderList.get(i).getDeadline() != null) {
-                    if (deadline.isEqual(taskReaderList.get(i).getDeadline()) ||
-                            (deadline.isBefore(taskReaderList.get(i).getDeadline()))) {
-                        taskReaderList.add(i, taskReader);
+            while (i < this.taskInfoList.size()) {
+                if (taskInfoList.get(i).getDeadline() != null) {
+                    if (deadline.isEqual(taskInfoList.get(i).getDeadline()) ||
+                            (deadline.isBefore(taskInfoList.get(i).getDeadline()))) {
+                        taskInfoList.add(i, taskInfo);
                         return;
                     }
                 } else {
-                    taskReaderList.add(i, taskReader);
+                    taskInfoList.add(i, taskInfo);
                     return;
                 }
                 i = i + 1;
             }
         }
-        taskReaderList.add(taskReader);
+        taskInfoList.add(taskInfo);
     }
 
     /***
-     * Update TaskInfoList to make it consistent with taskReaderList
+     * Update viewInfoList to make it consistent with taskReaderList
```

Comment:
to make it consistent with taskIntoList

Reply:
right good catch. i will update it in another pr.